kv_cache_manager.py

```
@@ -0,0 +1,795 @@
+        persist_path.parent.mkdir(parents=True, exist_ok=True)
+
+        with self.lock:
+            try:
+                data = {
+                    "config": self.config.to_dict(),
+                    "entries": {k: v.to_dict() for k, v in self.entries.items()},
+                    "bitmap": self.allocator.to_bytes().hex(),
+                    "data": self._data.hex(),
+                }
+                persist_path.write_text(json.dumps(data))
+                return True
```

Ruff (0.14.7)

Consider moving this statement to an `else` block

(TRY300)